pipelines/skylab/slideseq/SlideSeq.wdl

```diff
@@ -0,0 +1,176 @@
+        String read_structure
+
+        File tar_star_reference
+        File annotations_gtf
+
+        String output_bam_basename
+        Boolean count_exons
+        File bead_locations
+
+    }
+
+    parameter_meta {
+        r1_fastq: "Array of Read 1 FASTQ files; forward read; contains cell barcodes and molecule barcodes"
+        r2_fastq: "Array of Read 2 FASTQ files; reverse read; contains cDNA fragment generated from captured mRNA"
+        i1_fastq: "Optional array of i1 FASTQ files; index read used for demultiplexing of multiple samples on one flow cell"
+        input_id: "Name of sample matching this file; inserted into read group header"
+        read_structure: "String used to specify the UMI (M) and Barcode (C) positions in the Read 1 FASTQ"
+    }
+
+    call Metrics.FastqMetricsSlideSeq as FastqMetrics {
+        input:
+            r1_fastq = r1_fastq,
+            read_structure = read_structure,
+            sample_id = input_id,
+            whitelist = bead_locations
+    }
+    call FastqProcessing.FastqProcessingSlidSeq as SplitFastq {
+        input:
+            r1_fastq = r1_fastq,
+            r2_fastq = r2_fastq,
+            i1_fastq = i1_fastq,
+            read_structure = read_structure,
+            sample_id = input_id,
+            whitelist = bead_locations
+    }
+    scatter(idx in range(length(SplitFastq.fastq_R1_output_array))) {
+        call StarAlign.STARsoloFastqSlideSeq as STARsoloFastqSlideSeq {
+            input:
+                r1_fastq = [SplitFastq.fastq_R1_output_array[idx]],
+                r2_fastq = [SplitFastq.fastq_R2_output_array[idx]],
+                whitelist = bead_locations,
+                tar_star_reference = tar_star_reference,
+                output_bam_basename = output_bam_basename + "_" + idx,
+                read_structure = read_structure,
+                count_exons = count_exons
+        }
+    }
+    call Merge.MergeSortBamFiles as MergeBam {
+        input:
+            bam_inputs = STARsoloFastqSlideSeq.bam_output,
+            output_bam_filename = output_bam_basename + ".bam",
+            sort_order = "coordinate"
+    }
+    call Metrics.CalculateGeneMetrics as GeneMetrics {
+        input:
+            bam_input = MergeBam.output_bam,
+            input_id = input_id
+    }
+    call Metrics.CalculateUMIsMetrics as UMIsMetrics {
+        input:
+            bam_input = MergeBam.output_bam,
+            input_id = input_id
+    }
+
+    call Metrics.CalculateCellMetrics as CellMetrics {
+        input:
+            bam_input = MergeBam.output_bam,
+            original_gtf = annotations_gtf,
+            input_id = input_id
+    }
+
+    call StarAlign.MergeStarOutput as MergeStarOutputs {
+        input:
+            barcodes = STARsoloFastqSlideSeq.barcodes,
+            features = STARsoloFastqSlideSeq.features,
+            matrix = STARsoloFastqSlideSeq.matrix,
+            input_id = input_id
+    }
+    if ( !count_exons ) {
```

Comment:
@khajoue2 It looks like the bead barcodes are being added to the Loom if the count exons option is false? Do we want the pipeline to behave like this or do we want to keep the Loom consistent with the count_exons = true option?